pyaedt/edb_core/edb_data/siwave_simulation_setup_data.py

```diff
@@ -246,130 +358,286 @@ def mesh_frequency(self):
         self._parent._update_setup()
 
     @include_vi_sources.setter
     def include_vi_sources(self, value):
-        self.sim_setup_info.SimulationSettings.UseCustomSettings = True
-        self.sim_setup_info.SimulationSettings.AdvancedSettings.IncludeVISources = value
+        edb_setup_info = self.sim_setup_info
+
+        edb_setup_info.SimulationSettings.AdvancedSettings.IncludeVISources = value
+        self._parent._edb_object = self._parent._set_edb_setup_info(edb_setup_info)
         self._parent._update_setup()
 
     @infinite_ground_location.setter
     def infinite_ground_location(self, value):
-        self.sim_setup_info.SimulationSettings.UseCustomSettings = True
-        self.sim_setup_info.SimulationSettings.AdvancedSettings.InfGndLocation = value
+        edb_setup_info = self.sim_setup_info
+
+        edb_setup_info.SimulationSettings.AdvancedSettings.InfGndLocation = value
+        self._parent._edb_object = self._parent._set_edb_setup_info(edb_setup_info)
         self._parent._update_setup()
 
     @max_coupled_lines.setter
     def max_coupled_lines(self, value):
-        self.sim_setup_info.SimulationSettings.UseCustomSettings = True
-        self.sim_setup_info.SimulationSettings.AdvancedSettings.MaxCoupledLines = value
+        edb_setup_info = self.sim_setup_info
+
+        edb_setup_info.SimulationSettings.AdvancedSettings.MaxCoupledLines = value
+        self._parent._edb_object = self._parent._set_edb_setup_info(edb_setup_info)
         self._parent._update_setup()
 
     @automatic_mesh.setter
     def automatic_mesh(self, value):
-        self.sim_setup_info.SimulationSettings.UseCustomSettings = True
-        self.sim_setup_info.SimulationSettings.AdvancedSettings.MeshAutoMatic = value
+        edb_setup_info = self.sim_setup_info
+
+        edb_setup_info.SimulationSettings.AdvancedSettings.MeshAutoMatic = value
+        self._parent._edb_object = self._parent._set_edb_setup_info(edb_setup_info)
         self._parent._update_setup()
 
     @perform_erc.setter
     def perform_erc(self, value):
-        self.sim_setup_info.SimulationSettings.AdvancedSettings.PerformERC = value
+        edb_setup_info = self.sim_setup_info
+        edb_setup_info.SimulationSettings.AdvancedSettings.PerformERC = value
+        self._parent._edb_object = self._parent._set_edb_setup_info(edb_setup_info)
         self._parent._update_setup()
 
     @mesh_frequency.setter
     def mesh_frequency(self, value):
-        self.sim_setup_info.SimulationSettings.UseCustomSettings = True
-        self.sim_setup_info.SimulationSettings.AdvancedSettings.MeshFrequency = value
+        edb_setup_info = self.sim_setup_info
+
+        edb_setup_info.SimulationSettings.AdvancedSettings.MeshFrequency = value
+        self._parent._edb_object = self._parent._set_edb_setup_info(edb_setup_info)
         self._parent._update_setup()
 
 
-class SiwaveDCAdvancedSettings(object):
+class DCSettings(SettingsBase):
```

Comment:
A "Parameters" section should also be added for this class, describing all parameters in the ``__init__`` method.